### PR TITLE
Some general improvements

### PR DESCRIPTION
**Pull request by DarkWiiPlayer**

### CODE REVIEW

File: lualog/util.lua [hunk 1]
```diff
@@ -12,36 +16,67 @@ function util.table.includes(t,element)
...
 end
 
+--- Returns whether an object is an instance of a certain class.
+-- @param instance The object to check
+-- @param class The class the object might be an instance of
+-- @treturn boolean Whether the object is an instance of the class
 function util.is_instance_of(instance,class)
-    return instance.__class_name == class
+    if not type(instance)=='table' then error("Argument #1 not an object (expected table, got "..type(instance)..")", 2) end
+    return and instance.__class_name == class
```
Comment (Desvelao) [inline lualog/util.lua]: Remove `and` after return statement. Raise an error this line.

**Review by Desvelao — CHANGES_REQUESTED** [review]

**Review by Desvelao — CHANGES_REQUESTED** [review]
Fix typos (if you can) in function docs:
- `@tparam` to `@param`
- `@treturn` to `@return`

Thank you for your contribution!

**Comment by DarkWiiPlayer** [comment]
> Fix typos (if you can) in function docs

`@treturn` and `@tparam` are special [LDoc tags][ldoc] for typed parameters and return values.

Unless you want compatibility with [LuaDoc (Which doesn't know these tags)][luadoc-tags], they are preferable in this case because they specify the the type of the parameter / return value in addition to its description.

[ldoc]: https://stevedonovan.github.io/ldoc/manual/doc.md.html "LDoc"
[luadoc-tags]: http://keplerproject.github.io/luadoc/manual.html#tags "LuaDoc Tags"

**Comment by Desvelao** [comment]
> > Fix typos (if you can) in function docs
> 
> `@treturn` and `@tparam` are special [LDoc tags](https://stevedonovan.github.io/ldoc/manual/doc.md.html) for typed parameters and return values.
> 
> Unless you want compatibility with [LuaDoc (Which doesn't know these tags)](http://keplerproject.github.io/luadoc/manual.html#tags), they are preferable in this case because they specify the the type of the parameter / return value in addition to its description.

Ok, it's perfect then.

**Comment by DarkWiiPlayer** [comment]
I'm actually starting to wonder how much sense it really makes to have the `Inspect` class when all it really does is one thing. Wouldn't it just be simpler to have a plain function instead?

**Comment by Desvelao** [comment]
> I'm actually starting to wonder how much sense it really makes to have the `Inspect` class when all it really does is one thing. Wouldn't it just be simpler to have a plain function instead?

Yes, i guess it would be better. Maybe remove `inpect.lua` file and move the function to `init.lua` as Lualog:inspect(t, options). It would be interesting return self of function to can chain methods.

```lua
function Lualog:inspect(t,options)
   print( inspect_table_function(t, options) ) -- options need to be configured to accept one-time options or self inspect options 
   return self -- to chain methods i mean
end
```

I don't know if it would be interesting accept vararg tables to inspect.

Feel free to try your solution.

**Comment by DarkWiiPlayer** [comment]
I think for now I'm done with the "general" improvements. I may have a go at turning inspect into a single function, though it's not like there's anything wrong with the object-oriented approach either.

Feel free to review and merge ♥

**Comment by Desvelao** [comment]
Perfect! Thank you! ;)

**Pull request merged**